### PR TITLE
CS1 videos UX for students

### PR DESCRIPTION
**Pull request by shubhi1092**
- Changed 'view levels' link to 'view map' on Student Dashboard for all courses
- Added 'view videos' link to Student Dashboard for Introduction to CS only
- Added a new view 'CourseVideosView' for showing CS1 videos

### CODE REVIEW

File: test/app/views/courses/CoursesView.spec.coffee [hunk 1]
```diff
@@ -1,5 +1,6 @@
 CoursesView = require 'views/courses/CoursesView'
 HeroSelectModal = require 'views/courses/HeroSelectModal'
+# Levels = require 'collections/Levels'
```
Comment (AndrewJakubowicz) [inline test/app/views/courses/CoursesView.spec.coffee]: Required?

File: app/views/courses/CoursesView.coffee [hunk 2]
```diff
@@ -241,3 +245,10 @@ module.exports = class CoursesView extends RootView
     courseID = $(e.target).data('course-id')
     window.tracker?.trackEvent 'Students View To Student Assessments View', category: 'Students', classroomID: classroomID, ['Mixpanel']
     application.router.navigate("/students/assessments/#{classroomID}##{courseID}", { trigger: true })
+  
+  onClickViewVideosLink: (e) ->
+    classroomID = $(e.target).data('classroom-id')
```
Comment (AndrewJakubowicz) [inline app/views/courses/CoursesView.coffee]: What do we think about storing attributes on the DOM element like this?

My minor concern/thoughts are that it maybe introduces an implicit dependency.
It's not explicit to trace the anchor tag + its attributes to the routing.

This appears to be the way it's been done so I think it's fine. I wonder if there is an alternative way.

File: test/app/views/courses/CoursesView.spec.coffee [hunk 3]
```diff
@@ -31,3 +32,35 @@ describe 'CoursesView', ->
       expect(view.openModalView).toHaveBeenCalled()
       args = view.openModalView.calls.argsFor(0)
       expect(args[0] instanceof HeroSelectModal).toBe(true)
+
+
+  # TODO: this test case is in progress
```
Comment (AndrewJakubowicz) [inline test/app/views/courses/CoursesView.spec.coffee]: Looking forward to these :)

**Comment by AndrewJakubowicz** [comment]
Not sure about requirements, but page could be optimized better for medium and small screens.

File: test/app/views/courses/CoursesView.spec.coffee [hunk 4]
```diff
@@ -30,4 +31,4 @@ describe 'CoursesView', ->
       view.$('.change-hero-btn').click()
       expect(view.openModalView).toHaveBeenCalled()
       args = view.openModalView.calls.argsFor(0)
-      expect(args[0] instanceof HeroSelectModal).toBe(true)
+      expect(args[0] instanceof HeroSelectModal).toBe(true)
```
Comment (AndrewJakubowicz) [inline test/app/views/courses/CoursesView.spec.coffee]: My phone is showing a change here that I can't see.  Probably nothing.

File: test/app/views/courses/CoursesView.spec.coffee (hunk 1 above)
Comment (shubhi1092) [reply test/app/views/courses/CoursesView.spec.coffee]: Nope, forgot to remove it.

File: test/app/views/courses/CoursesView.spec.coffee (hunk 4 above)
Comment (shubhi1092) [reply test/app/views/courses/CoursesView.spec.coffee]: I think just a `\n` change probably.

**Comment by shubhi1092** [comment]
The videos are allowed to be embedded only on domain 'codecombat.com', so it doesnt appear locally. But I tested it once with that setting and it worked fine.

**Pull request merged**